src/scenic/syntax/scenic.gram

```diff
@@ -1441,11 +1441,11 @@ scenic_implication (memo):
     | disjunction
 
 disjunction (memo):
-    | a=conjunction b=('or' c=conjunction { c })+ { ast.BoolOp(op=ast.Or(), values=[a] + b, LOCATIONS) }
+    | a=conjunction b=('or' c=(scenic_temporal_prefix | conjunction) { c })+ { ast.BoolOp(op=ast.Or(), values=[a] + b, LOCATIONS) }
```

Comment:
This change allows `scenic_temporal_prefix` to appear in the second operand and onward of `or`. The temporal prefix operators cannot appear in the first operand because `always x or y` should be parsed as `always (x or y)` and that part is already done by the rules above.

This results in ambiguous grammar because `x or always y or z` can be parsed as either `x or (always y or z)` (this is what we want) and `x or (always y) or z` (not what we want, but there is a derivation). However, Pegen uses whatever matches first, so in this case, we get the former. I added test cases to make sure the grammar can handle this case.